### PR TITLE
optimize: use curator instead of zkclient in config model

### PR DESCRIPTION
**Pull request by leizhiyuan**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

fiees #6772
### Ⅳ. Describe how to verify it

test case

### CODE REVIEW

**Review by funky-eyes — COMMENTED** [review]
Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project seata-config-zk: Compilation failure: Compilation failure: 
Error:  /home/runner/work/incubator-seata/incubator-seata/config/seata-config-zk/src/test/java/org/apache/seata/config/zk/ZkConfigurationTest.java:[21,31] package org.apache.curator.test does not exist
Error:  /home/runner/work/incubator-seata/incubator-seata/config/seata-config-zk/src/test/java/org/apache/seata/config/zk/ZkConfigurationTest.java:[35,22] cannot find symbol
Error:    symbol:   class TestingServer
Error:    location: class org.apache.seata.config.zk.ZkConfigurationTest
Error:  /home/runner/work/incubator-seata/incubator-seata/config/seata-config-zk/src/test/java/org/apache/seata/config/zk/ZkConfigurationTest.java:[41,22] cannot find symbol
Error:    symbol:   class TestingServer
Error:    location: class org.apache.seata.config.zk.ZkConfigurationTest
Error:  -> [Help 1]

File: config/seata-config-zk/pom.xml [hunk 1]
```diff
@@ -45,6 +45,18 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>org.apache.curator</groupId>
+            <artifactId>curator-recipes</artifactId>
```
Comment (funky-eyes) [inline config/seata-config-zk/pom.xml:40]: 按目前的实现，zkclient是不是可以不要了？
According to the current implementation, can zkclient be replaced?

**Comment by funky-eyes** [comment]
@leizhiyuan Please fix the ci problem.

**Review by funky-eyes — COMMENTED** [review]
https://github.com/apache/incubator-seata/blob/2.x/changes/zh-cn/2.x.md
https://github.com/apache/incubator-seata/blob/2.x/changes/en-us/2.x.md
请在上述md文件中登记pr信息和作者信息
Please register pr information and author information in the above md file

**Comment by funky-eyes** [comment]
Error:  Failures: 
Error:    ZkConfigurationTest.testPutConfig:83 expected: <true> but was: <false>
[INFO] 
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0
[INFO]

**Comment by leizhiyuan** [comment]
> Error: Failures: Error: ZkConfigurationTest.testPutConfig:83 expected: but was: [INFO] Error: Tests run: 3, Failures: 1, Errors: 0, Skipped: 0 [INFO]

It seems to have something to do with the environment.,Local mac has always passed.,I'll find a Linux machine to run it.

image

**Comment by leizhiyuan** [comment]
image

run ok in linux。。

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**